docs/configuration_files.rst

```diff
@@ -0,0 +1,341 @@
+###################
+Configuration files
+###################
+
+This page outlines how to construct configuration files to run your own routines
+with `~skypy.pipeline.Pipeline`.
+
+`SkyPy` is an astrophysical simulation pipeline tool that allows to define any
+arbitrary workflow and store data in table format. You may use `SkyPy` `~skypy.pipeline.Pipeline`.
+to call any function --your own, from any compatible external software or from the `SkyPy library`.
+Then `SkyPy` deals with the data dependencies and provides a library of functions to be used with it.
+
+These guidelines start with an example using one of the `SkyPy` functions, and it follows
+the concrete YAML syntax necessary for you to write your own configuration files, beyond using `SkyPy`
+functions.
+
+SkyPy example
+-------------
+
+In this section, we exemplify how you can write a configuration file and use some of the `SkyPy` functions.
+In this example, we sample redshifts and magnitudes from the SkyPy luminosity function, `~skypy.galaxies.schechter_lf`.
+
+- `Define variables`:
+
+From the documentation, the parameters for the `~skypy.galaxies.schechter_lf` function are: ``redshift``, the characteristic absolute magnitude ``M_star``, the amplitude ``phi_star``, faint-end slope parameter ``alpha``,
+the magnitude limit ``magnitude_limit``, the fraction of sky ``sky_area``, ``cosmology`` and ``noise``.
+If you are planning to reuse some of these parameters, you can define them at the top-level of your configuration file.
+In our example, we are using ``Astropy`` linear and exponential models for the characteristic absolute magnitude and the amplitude, respectively.
+Also, ``noise`` is an optional parameter and you could use its default value by omitting its definition.
+
+  .. code:: yaml
+
+    cosmology: !astropy.cosmology.default_cosmology.get
+    z_range: !numpy.linspace [0, 2, 21]
+    M_star: !astropy.modeling.models.Linear1D [-0.9, -20.4]
+    phi_star: !astropy.modeling.models.Exponential1D [3e-3, -9.7]
+    magnitude_limit: 23
+    sky_area: 10 deg2
+
+- `Tables and columns`:
+
+You can create a table ``blue_galaxies`` and for now add the columns for redshift and magnitude (note here the ``schechter_lf`` returns a 2D object)
+
+  .. code:: yaml
+
+    tables:
+      blue_galaxies:
+        redshift, magnitude: !skypy.galaxies.schechter_lf
+      		redshift: $z_range
+      		M_star: $M_star
+      		phi_star: $phi_star
+      		alpha: -1.3
+      		m_lim: $magnitude_limit
+      		sky_area: $sky_area
+
+`Important:` if cosmology is detected as a parameter but is not set, it automatically uses the cosmology variable defined at the top-level of the file.
+
+This is how the entire configuration file looks like!
+You may now save it as ``luminosity.yml`` and run it using the `SkyPy` `~skypy.pipeline.Pipeline`:
+
+.. literalinclude:: luminosity.yml
+   :language: yaml
+
+You may now save it as ``luminosity.yml`` and run it using the `SkyPy` `~skypy.pipeline.Pipeline`:
```

Comment:
This sentence is twice. Also in line 59. Is this on purpose?
I think we want to keep this line and remove 59?